README.md

```diff
@@ -158,6 +158,39 @@ And here's an example of how you might document a class:
      ) { }
 ```
 
+Puppet providers often rely on different types of metaprogramming to create
+parameters and methods. Since Strings does not execute the code which is being
+documented you must provide hints on how to document your code. To document a
+parameter which is automatically created you must use the special directive
+`@!puppet.provider.param` Which may take types, the parameter name, and a
+description.
+
+Strings will automatically extract the `@doc` provider docstring and any `desc`
+parameter docstrings.
+
+```ruby
+# @!puppet.provider.param my_parameter This parameter needs to be explicitly
```

Comment:
Right on! I didn't pick up on that from the words above, but now that you said that, I do from rereading it. Perhaps you should mention the second part first. I word smithed this a little to help clarify.

~~~md
Strings will automatically extract the `@doc` provider docstring and any `desc` parameter/property docstrings.

Sometimes however, Puppet providers use metaprogramming to create parameters and methods automatically. In those cases Strings will not be able to document them automatically (Strings doesn't execute the code that would generate those parameters), so you will need to provide hints on how to document your code. To document a parameter which is automatically created you must use the special directive `@!puppet.provider.param` Which may take types, the parameter name, and a description.
~~~

Does  "parameter which is automatically created" also apply to properties that are auto created? I assume yes. I also don't know how often one could create a property automatically as the work involved in those is more difficult.